### PR TITLE
Remove unsupported zone pivots (serialization)

### PR DESCRIPTION
**Pull request by gewarren**
- Remove unsupported zone pivots (3.1 and 5)
- Remove zones where they're now no longer needed, i.e. they're the only zone left
- Also use full display name of STJ namespace.

I also added a [tag](https://github.com/dotnet/docs/releases/tag/3.1-zone-pivot-removal) to the repo so we can retrieve anything that might be needed in the future.

Contributes to #35434 

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/deploying/trimming/trimming-options.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/deploying/trimming/trimming-options.md) | [Trimming options](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options?branch=pr-en-us-35439) |
| [docs/core/extensions/queue-service.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/extensions/queue-service.md) | [Create a Queue Service](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/queue-service?branch=pr-en-us-35439) |
| [docs/core/extensions/scoped-service.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/extensions/scoped-service.md) | [Use scoped services within a `BackgroundService`](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/scoped-service?branch=pr-en-us-35439) |
| [docs/core/extensions/timer-service.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/extensions/timer-service.md) | [docs/core/extensions/timer-service](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/timer-service?branch=pr-en-us-35439) |
| [docs/core/extensions/windows-service.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/extensions/windows-service.md) | [Create Windows Service using `BackgroundService`](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/windows-service?branch=pr-en-us-35439) |
| [docs/core/extensions/workers.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/extensions/workers.md) | [Worker Services in .NET](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/workers?branch=pr-en-us-35439) |
| [docs/core/tutorials/debugging-with-visual-studio-code.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/debugging-with-visual-studio-code.md) | [Debug a .NET console application using Visual Studio Code](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/debugging-with-visual-studio-code?branch=pr-en-us-35439) |
| [docs/core/tutorials/library-with-visual-studio-code.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/library-with-visual-studio-code.md) | [Tutorial: Create a .NET class library using Visual Studio Code](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/library-with-visual-studio-code?branch=pr-en-us-35439) |
| [docs/core/tutorials/library-with-visual-studio.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/library-with-visual-studio.md) | [docs/core/tutorials/library-with-visual-studio](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/library-with-visual-studio?branch=pr-en-us-35439) |
| [docs/core/tutorials/publishing-with-visual-studio-code.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/publishing-with-visual-studio-code.md) | [Tutorial: Publish a .NET console application using Visual Studio Code](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/publishing-with-visual-studio-code?branch=pr-en-us-35439) |
| [docs/core/tutorials/publishing-with-visual-studio.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/publishing-with-visual-studio.md) | [Publish a .NET console application using Visual Studio](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/publishing-with-visual-studio?branch=pr-en-us-35439) |
| [docs/core/tutorials/testing-library-with-visual-studio-code.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/testing-library-with-visual-studio-code.md) | [Tutorial: Test a .NET class library using Visual Studio Code](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/testing-library-with-visual-studio-code?branch=pr-en-us-35439) |
| [docs/core/tutorials/testing-library-with-visual-studio.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/testing-library-with-visual-studio.md) | [docs/core/tutorials/testing-library-with-visual-studio](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/testing-library-with-visual-studio?branch=pr-en-us-35439) |
| [docs/core/tutorials/with-visual-studio-code.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/with-visual-studio-code.md) | [Tutorial: Create a .NET console application using Visual Studio Code](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/with-visual-studio-code?branch=pr-en-us-35439) |
| [docs/core/tutorials/with-visual-studio.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/core/tutorials/with-visual-studio.md) | [Create a .NET console application using Visual Studio](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/with-visual-studio?branch=pr-en-us-35439) |
| [docs/csharp/asynchronous-programming/start-multiple-async-tasks-and-process-them-as-they-complete.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/csharp/asynchronous-programming/start-multiple-async-tasks-and-process-them-as-they-complete.md) | [Process asynchronous tasks as they complete (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/start-multiple-async-tasks-and-process-them-as-they-complete?branch=pr-en-us-35439) |
| [docs/standard/datetime/system-text-json-support.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/datetime/system-text-json-support.md) | [DateTime and DateTimeOffset support in System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/datetime/system-text-json-support?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/configure-options.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/configure-options.md) | [docs/standard/serialization/system-text-json/configure-options](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/configure-options?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/converters-how-to.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/converters-how-to.md) | [How to write custom converters for JSON serialization (marshalling) in .NET](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/converters-how-to?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/customize-properties.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/customize-properties.md) | [How to customize property names and values with System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/customize-properties?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/handle-overflow.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/handle-overflow.md) | [How to handle overflow JSON or use JsonElement or JsonNode in System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/handle-overflow?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/how-to.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/how-to.md) | [docs/standard/serialization/system-text-json/how-to](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/ignore-properties.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/ignore-properties.md) | [How to ignore properties with System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/ignore-properties?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/immutability.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/immutability.md) | [How to use immutable types and non-public accessors with System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/immutability?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/invalid-json.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/invalid-json.md) | [How to allow some kinds of invalid JSON with System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/invalid-json?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md) | [docs/standard/serialization/system-text-json/migrate-from-newtonsoft](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/preserve-references.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/preserve-references.md) | [How to preserve references and handle or ignore circular references in System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/preserve-references?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/source-generation-modes.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/source-generation-modes.md) | [How to choose reflection or source generation in System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation-modes?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/supported-collection-types.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/supported-collection-types.md) | [Supported collection types in System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/supported-collection-types?branch=pr-en-us-35439) |
| [docs/standard/serialization/system-text-json/use-dom-utf8jsonreader-utf8jsonwriter.md](https://github.com/dotnet/docs/blob/804101d8005675e7b7a30509d7ef14e30cc9d63d/docs/standard/serialization/system-text-json/use-dom-utf8jsonreader-utf8jsonwriter.md) | [How to use a JSON document, Utf8JsonReader, and Utf8JsonWriter in System.Text.Json](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/use-dom-utf8jsonreader-utf8jsonwriter?branch=pr-en-us-35439) |

</details>

> **Note**
> This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/docs/pull/35439/checks).

### CODE REVIEW

**Comment by adegeo** [comment]
Regarding the pivot id change, the guidance (when I last looked a few years ago) was to explicitly version the ID and don't use a generic catch all pivot that can change over time. If there is something that specifically required 6-7 and 8 comes along and 6 is dropped, you should create a new pivot id for 7-8.

**Comment by gewarren** [comment]
> Regarding the pivot ID change, the guidance (when I last looked a few years ago) was to explicitly version the ID and don't use a generic catch all pivot that can change over time. If there is something that specifically required 6-7 and 8 comes along and 6 is dropped, you should create a new pivot id for 7-8.

How about if I name it "supported-dotnet-versions" and then update the versions under it as versions come and go (out of support)?

**Review by tdykstra — APPROVED** [review]
LGTM, one comment.

File: docs/standard/serialization/system-text-json/use-dom-utf8jsonreader-utf8jsonwriter.md [hunk 1]
```diff
@@ -127,13 +112,11 @@ Here's an example of the JSON that this code processes:
...
 :::code language="json" source="snippets/system-text-json-how-to/csharp/GradesPrettyPrint.json":::
 
-:::zone pivot="dotnet-7-0,dotnet-6-0"
 For a similar example that uses `JsonNode` instead of `JsonDocument`, see [JsonNode average grade example](#jsonnode-average-grade-example).
-:::zone-end
 
 ### How to search a JsonDocument and JsonElement for sub-elements
 
-Searches on `JsonElement` require a sequential search of the properties and hence are relatively slow (for example when using `TryGetProperty`). <xref:System.Text.Json> is designed to minimize initial parse time rather than lookup time. Therefore, use the following approaches to optimize performance when searching through a `JsonDocument` object:
+Searches on `JsonElement` require a sequential search of the properties and hence are relatively slow (for example when using `TryGetProperty`). <xref:System.Text.Json?displayProperty=fullName> is designed to minimize initial parse time rather than lookup time. Therefore, use the following approaches to optimize performance when searching through a `JsonDocument` object:
```
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/standard/serialization/system-text-json/use-dom-utf8jsonreader-utf8jsonwriter.md:119]: Ah, yes I forgot about that. Is it okay to leave it in so it's clear that the whole thing is displayed to people that might not be familiar with the rules? I feel like it doesn't hurt anything.

**Pull request merged**